src/init/VarResolver.ts

```diff
@@ -0,0 +1,139 @@
+import yargs from 'yargs';
+import { AsyncHandler } from '../util/handlers/AsyncHandler';
+import { ensureTrailingSlash, modulePathPlaceholder, resolveAssetPath } from '../util/PathUtil';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+/**
+ * CLI options needed for performing meta-process of app initialization.
+ * These options doesn't contribute to components-js vars normally.
+ */
+export const baseYargsOptions: Record<string, yargs.Options> = {
+  config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+  loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
+  mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+  varConfig: { type: 'string', alias: 'v', default: defaultVarConfig, requiresArg: true },
+};
+
+/**
+ * A handler that takes args, and returns computed variable value
+ */
+export abstract class VarComputer extends AsyncHandler<yargs.Arguments, any> {
+  public abstract handle(args: yargs.Arguments): Promise<any>;
+}
+
+// Currently componentsjs-builder giving error, if we use external TSTypeReference.
+// Re-definition is temporary work around.
+export interface IYargsOptions extends yargs.Options {
```

Comment:
```suggestion
export interface YargsOptions extends yargs.Options {
```

And similar below. We don't use `I` to indicate interfaces. The only interfaces where this is the case is from external libraries/typings.